packages/network_partitions/src/leiden/neighboring_clusters.rs

```diff
@@ -11,26 +11,23 @@ pub struct NeighboringClusters {
 
 impl NeighboringClusters {
     pub fn with_capacity(length: usize) -> NeighboringClusters {
-        return NeighboringClusters {
+        NeighboringClusters {
             neighboring_clusters: Vec::with_capacity(length + 1),
             neighbor_edge_weights_within_cluster: vec![f64::NAN; length + 1],
             current_cluster: None,
-        };
+        }
     }
 
     pub fn reset_for_current_cluster(
         &mut self,
         current_cluster: usize,
     ) {
-        match self.current_cluster {
-            Some(current_cluster) => {
-                self.neighbor_edge_weights_within_cluster[current_cluster] = f64::NAN;
-                for cluster in &self.neighboring_clusters {
-                    self.neighbor_edge_weights_within_cluster[*cluster] = f64::NAN;
-                }
-                self.neighboring_clusters.clear();
+        if let Some(current_cluster) = self.current_cluster {
```

Comment:
It's been a minute since I've written any Rust, but this looks like an assignment in an `if` expression, and I just wanted to confirm your intent here.

Reply:
I *believe* it is - I think self.current_cluster can be None, and rather than try to match on a Some(current_cluster) and handle the None in said match when I don't actually want to do anything if it *is* None, I fabricate a Some(usize) and test equality. I don't know why I thought this was strictly better than the other way, it has been a few years at this point. Currently I have zero strong opinions about it